cmd/posix.go

```diff
@@ -1358,9 +1358,18 @@ func (s *posix) RenameFile(srcVolume, srcPath, dstVolume, dstPath string) (err e
 		if err == nil && !isDirEmpty(dstFilePath) {
 			return errFileAccessDenied
 		}
-		if !os.IsNotExist(err) {
+		if err != nil && !os.IsNotExist(err) {
 			return err
 		}
+		// Empty destination remove it before rename.
+		if isDirEmpty(dstFilePath) {
```

Comment:
Can you say more about this change ? what particular case does it fix ?

Reply:
It fixes an important issue #7421, this perhaps a really old bug. I can't write more because it's already written above why when srcIsDir destination should be empty.